src/llmq/quorums_signing_shares.cpp

```diff
@@ -1012,49 +1062,88 @@ bool CSigSharesManager::SendMessages()
 
         auto it1 = sigSessionAnnouncements.find(pnode->id);
         if (it1 != sigSessionAnnouncements.end()) {
+            std::vector<CSigSesAnn> msgs;
+            msgs.reserve(it1->second.size());
             for (auto& sigSesAnn : it1->second) {
                 LogPrint("llmq", "CSigSharesManager::SendMessages -- QSIGSESANN signHash=%s, sessionId=%d, node=%d\n",
                          CLLMQUtils::BuildSignHash(sigSesAnn).ToString(), sigSesAnn.sessionId, pnode->id);
-                g_connman->PushMessage(pnode, msgMaker.Make(NetMsgType::QSIGSESANN, sigSesAnn), false);
+                msgs.emplace_back(sigSesAnn);
+                if (msgs.size() >= MAX_MSGS_CNT_QSIGSESANN) {
```

Comment:
Sending with `size > max` is a recipe for a ban :) It should never happen given the way it's written now but it looks confusing imo, should really be
```suggestion
                if (msgs.size() == MAX_MSGS_CNT_QSIGSESANN) {
```

Same for QGETSIGSHARES and QSIGSHARESINV.

Reply:
Yepp, agree. Fixed.